.github/cloud-samples-tools/pkg/config/config.go

```diff
@@ -61,30 +63,22 @@ func (c *Config) Save(file *os.File) error {
 
 // LoadConfig loads the config from the given path.
 func LoadConfig(path string) (*Config, error) {
-	// Read the JSONC file.
-	sourceJsonc, err := os.ReadFile(path)
-	if err != nil {
-		return nil, err
+	// Set the config default values.
+	config := Config{
+		Match: []string{"*"},
 	}
 
-	// Strip the comments and load the JSON.
-	sourceJson := multiLineCommentsRegex.ReplaceAll(sourceJsonc, []byte{})
-	sourceJson = singleLineCommentsRegex.ReplaceAll(sourceJson, []byte{})
-
-	var config Config
-	err = json.Unmarshal(sourceJson, &config)
+	// This mutates `config` so there's no need to reassign it.
+	// It keeps the default values if they're not in the JSON file.
+	err := readJsonc(path, &config)
 	if err != nil {
 		return nil, err
 	}
 
-	// Set default values if they are not set.
+	// Validate for required values.
 	if config.PackageFile == nil {
 		return nil, errors.New("package-file is required")
 	}
-	if config.Match == nil {
```

Comment:
I *think* nil vs. empty slices are technically different in Go. [Go Playground link](https://go.dev/play/p/FyKK2w8D4U0).  And the Json parser could return either depending on what is in the underlying file. 

Consider `if len(config.Match) == 0 {`, which covers both cases. (If changes, this pattern probably needs updates in multiple places. Also might make sense to do as a followup PR. 

This is not an area where I'm an expert. Calling @telpirion for an additional look.

Reply:
Interesting, yet another pitfall. I did change the way I passed default values to the JSON parser, so this is no longer the case. I found a simpler way.

Reply:
That is a good catch! I agree, the `len()` function should be used here.